astropy/io/fits/tests/test_connect.py

```diff
@@ -329,3 +329,13 @@ def test_unit_warnings_read_write(tmpdir):
     with catch_warnings() as l:
         Table.read(filename, hdu=1)
     assert len(l) == 0
+
+
+def test_convert_comment_convention(tmpdir):
+    """
+    Regression test for https://github.com/astropy/astropy/issues/6079
+    """
+    filename = DATA + '/table_with_comment.fits'
```

Comment:
Agree, we should just be able to first write a table with comments to a temp file and then read it out.  Unless there is some weird FITS file keyword format that `io.fits` cannot directly write?  Will `io.fits` work with a `StringIO` buffer?